vendor/views/vendor.py

```diff
@@ -308,7 +308,7 @@ class OrderHistoryListView(LoginRequiredMixin, ListView):
     def get_queryset(self):
         try:
             # The profile and user are site specific so this should only return what's on the site for that user excluding the cart
-            return self.request.user.customer_profile.get().invoices.filter(status__gt=Invoice.InvoiceStatus.CART)
+            return self.request.user.customer_profile.get(site=settings.SITE_ID).invoices.filter(status__gt=Invoice.InvoiceStatus.CART)
```

Comment:
You should user `get_current_site(self.request)` instead of setting.SITE_ID
import is:
`from django.contrib.sites.shortcuts import get_current_site`

Reply:
If it expects a Site object in the query (vs a string or int), this is the easiest way get it.